cli/src/main/java/com/devonfw/tools/ide/commandlet/UpgradeSettingsCommandlet.java

```diff
@@ -0,0 +1,366 @@
+      variable = variable.substring(variable.indexOf("(") + 1, variable.indexOf(")") - 1);
+      String url = variable.substring(variable.indexOf("https://"), variable.indexOf(" "));
+      String[] toolsArray = variable.split(" ");
+      for (String tool : toolsArray) {
+        JSONObject toolObject = new JSONObject();
+        String[] toolParts = tool.split(":");
+        toolObject.put("name", toolParts[0]);
+        toolObject.put("version", toolParts[1]);
+        if (toolParts.length > 2) {
+          toolObject.put("os-agnostic", toolParts[2].equals("all"));
+          toolObject.put("arch-agnostic", toolParts[2].equals("all"));
+        } else {
+          toolObject.put("os-agnostic", false);
+          toolObject.put("arch-agnostic", false);
+        }
+        tabelObject.add(toolObject);
+      }
+      JSONObject finalObject = new JSONObject();
+      finalObject.put("tools", tabelObject);
+      finalObject.put("url", url);
+
+      FileWriter writer = new FileWriter(context.getIdeHome().resolve("settings/custom-tools.json").toString());
+      writer.write(finalObject.toJSONString());
+
+      writer.flush();
+      writer.close();
+
+      context.success("Created custom-tools.json at: " + context.getIdeHome().resolve("settings/custom-tools.json"));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+
+    }
+  }
+
+  private void updateDevonProperties() {
+    List<Path> pathList = new ArrayList<>();
+    EnvironmentVariables devonPropertiesPath = context.getVariables();
+    while (devonPropertiesPath != null) {
+      if (devonPropertiesPath.getPropertiesFilePath() != null) {
+        pathList.add(devonPropertiesPath.getPropertiesFilePath().getParent().resolve("devon.properties"));
+      }
+      devonPropertiesPath = devonPropertiesPath.getParent();
+    }
+
+    for (Path filePath : pathList) {
+      if (!Files.exists(filePath)) {
+        continue;
+      }
+      Path target = filePath.getParent().resolve("ide.properties");
+      Properties devonProperties = new Properties();
+      devonProperties.put("IDE_VARIABLE_SYNTAX_LEGACY_SUPPORT_ENABLED", "false");
+      try {
+        List<String> readLines = Files.readAllLines(filePath);
+        String[] split;
+        for (String line : readLines) {
+          if (!line.contains("#") && !line.isEmpty()) {
+            if (line.contains("DEVON_IDE_CUSTOM_TOOLS")) {
+              createCustomToolsJson(line);
+              continue;
+            }
+            if (line.contains("DEVON_")) {
+              line = line.replace("DEVON_", "");
+            }
+            split = line.split("[ =]");
+            if (split.length == 3) {
+              devonProperties.put(split[1], new String[] { split[0], split[2] });
+            }
+            if (split.length == 2) {
+              devonProperties.put(split[0], split[1]);
+            }
+          }
+        }
+
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
You are duplicating the code to read a `devon.properties` and a `ide.properties` file.
https://github.com/devonfw/IDEasy/blob/76124f73b5008293ca845d94c65850c648e2a0d1/cli/src/main/java/com/devonfw/tools/ide/environment/EnvironmentVariablesPropertiesFile.java#L71-L86

We should use our existing infrastructure to do that.

Also your implementation contains some problems. E.g. `#` can also be used inside a value (e.g. to separate branch of a git-url) and I guess you are trying to ignore comments here. Using String "fiddling" with methods such as `contains` is also dangerous since it will also match in unexpected cases. Surely not a very relevant scenario but rather an odd edge-case but in case a user has `DISABLED_DEVON_IDE_CUSTOM_TOOLS=(...)` in his config, this will cause undesired effects (and I have seen users who were not aware that you can outcomment a line by adding `#` at the beginning but were simply adding `DISABLED_` as prefix). As you can see in the linked code we already have better infrastructure to parse a line and separate variable name and value. This should be reused.

Reply:
After re-reading all the code of this method I understand this was an important blind spot.
You tried to reimplement a lot of things that have already been solved in our code but could not find this.
Therefore, you ran into all these challenges how to deal with comments, how to split into property name and value, etc.